src/GoTrueClient.ts

```diff
@@ -251,6 +251,55 @@ export default class GoTrueClient {
     }
   }
 
+  /**
+   * Sets user data on session.
+   * @param access_token a jwt access token
+   */
+  async setUser(access_token: string): Promise<{ user: User | null; error: Error | null }> {
```

Comment:
one note: the discussion in #34 was centered around potential _synchronous_ solutions.

in a node context, we've extracted a known valid jwt access token from a cookie, and simply need to pass along the jwt in the `Authorization` header when interacting with supabase APIs. ideally, this would be possible without introducing unnecessary `async` data fetching overhead through calls to `api.getUser` / `api.refreshAccessToken`.

Reply:
agreed - the goal of this PR should be to set the `auth` headers. 

I have some time today, so I'll jump in here now and help out

Reply:
@kiwicopple so then what do you suggest?

Reply:
I'll test out your changes in a full `supabase-js` library on a node environment and let you know :)

Reply:
yeah that sounds awesome :fire: